src/chainparams.cpp

```diff
@@ -490,11 +490,8 @@ void CRegTestParams::UpdateActivationParametersFromArgs(const ArgsManager& args)
 {
     if (args.IsArgSet("-segwitheight")) {
         int64_t height = args.GetArg("-segwitheight", consensus.SegwitHeight);
-        if (height < -1 || height >= std::numeric_limits<int>::max()) {
-            throw std::runtime_error(strprintf("Activation height %ld for segwit is out of valid range. Use -1 to disable segwit.", height));
-        } else if (height == -1) {
-            LogPrintf("Segwit disabled for testing\n");
-            height = std::numeric_limits<int>::max();
```

Comment:
With the `segwitheight=-1` option removed, I think that also `IsScriptWitnessEnabled()` in `validation.cpp` has become obsolete and could be removed as well (the commentary there also seems to says that it's only kept along for `p2p_segwit.py`).

Reply:
There is another check for `consensusParams.SegwitHeight != std::numeric_limits<int>::max())` in `GenerateCoinbaseCommitment()`, also in validation.

Reply:
Great catches. Thank you, @mzumsande! Both comments addressed.